### PR TITLE
Introduced upgraded status js api according latest web3 opt-in api

### PR DESCRIPTION
**Pull request by flexsurfer**
According to the latest web3 opt-in EIP status injects read-only ethereum provider, we can expand this provider with status api, so there is no need in messages anymore

Before
```JavaScript
window.addEventListener('statusapi', function (event) {
    console.log(event.detail.permissions) //=> ["CONTACT_CODE"] if allowed 
    console.log(event.detail.data["CONTACT_CODE"]) //=> "0x0012300..123" if allowed
});
// request status API
window.postMessage({ type: 'STATUS_API_REQUEST', permissions: ["CONTACT_CODE"] }, '*');
```

After
```JavaScript
ethereum.status.getContactCode().then(data => {console.log('Contact code:', data)});
```

### CODE REVIEW

**Review by yenda — CHANGES_REQUESTED** [review]
the animation can show with a delay. we should never have to use dispatch and dispatch later in effects

https://github.com/status-im/status-react/pull/5824/files#diff-eeb5b0806bc1baaec4e6d3739c4a3681R17

File: resources/js/web3_init.js [hunk 1]
```diff
@@ -6,70 +6,77 @@ function bridgeSend(data){
...
-    {
-        var id = data.data.messageId;
-        var callback = callbacks[id];
-        if (callback) {
-            var result = data.result;
-            var regex = new RegExp(callback.regex);
-            if (regex.test(result)) {
-                if (callback.resolve) {
-                    callback.resolve(result);
+        } else if (data.type === "web3-send-async-callback") {
```
Comment (yenda) [inline resources/js/web3_init.js:35]: why can't everything be handled as an api response ?

**Comment by flexsurfer** [comment]
@yenda what is wrong with dispatching event?

**Comment by flexsurfer** [comment]
@yenda moved `dispatch` to UI

**Review by yenda — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

File: src/status_im/ui/screens/browser/permissions/views.cljs [hunk 2]
```diff
@@ -11,69 +11,125 @@
...
+  (anim/start
+   (anim/parallel
+    [(anim/spring bottom-anim-value {:toValue -20})
+     (anim/timing alpha-value {:toValue  0.6
+                               :duration 500})])))
+
+(defn permission-details [requested-permission]
+  (get browser.permissions/supported-permissions requested-permission))
+
+#_{:component-will-update (fn [this [_ _ {:keys [requested-permission]}]]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/browser/permissions/views.cljs]: @yenda do we need this?

**Review by goranjovic — APPROVED** [review]

**Comment by goranjovic** [comment]
@flexsurfer are there intentionally two commits?

**Comment by flexsurfer** [comment]
@goranjovic yes, one from Eric

**Comment by Serhy** [comment]
Looks good to me!

**Pull request merged**